pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -242,9 +242,13 @@ func (r *ReconcileMongoDB) Reconcile(request reconcile.Request) (reconcile.Resul
 	if instance.Spec.StorageClass == "" {
 		// TODO: weird because the storage class on OCP is opened for all
 		// Need to deploy an OCP cluster on AWS to verify
-		storageclass, err = r.getstorageclass()
-		if err != nil {
-			return reconcile.Result{}, err
+		if instance.Status.StorageClass == "" {
```

Comment:
I think we should move this out of the `if instance.Spec.StorageClass == ""` condition.

We should respect data, not user's action.

When user update the StorageClass in Spec, they need to remove the CR and re-create it, this can explicitly tell user that their data will be removed since they re-created the CR without a different storageclass.